Hackathon_Converter/MainForm.cs

```diff
@@ -51,63 +55,495 @@ private void OnAddFilesButtonClick(object sender, EventArgs e)
+        {
+            DataGridViewSelectedCellCollection SelectedFiles = fileDataGrid.SelectedCells;
+            int numRemove = 1;
+            if (fileDataGrid.Rows.Count > 1)
             {
-                var cell = row.Cells[0];
-                if (cell.Value == null)
+
+                
+                foreach(DataGridViewCell SelectFile in SelectedFiles)
                 {
-                    continue;
+                    if(SelectFile.OwningRow.Index != fileDataGrid.RowCount-1)
+                        fileDataGrid.Rows.Remove(SelectFile.OwningRow);
                 }
-                var filename = cell.Value.ToString();
-                var text = File.ReadAllText(filename);
-                usfm.Insert(parser.ParseFromString(text));
             }
-            var html = renderer.Render(usfm);
-            var bytes = Encoding.UTF8.GetBytes(html);
-            htmlStream.Write(bytes, 0, bytes.Length);
-            htmlStream.Close();
 
-            var htmlFilename = saveFileDialog.FileName;
-            var dirname = Path.GetDirectoryName(htmlFilename);
-            var cssFilename = Path.Combine(dirname, "style.css");
-            if (File.Exists(cssFilename) == false)
+            
+            if (fileDataGrid.Rows.Count == 1)
+                numRemove = 0;
+
+            btn_Remove.Text = $"Delete ({numRemove}) Files";
+
+
+        }
+
+        private void Btn_NewProj_Click(object sender, EventArgs e)
+        {
+            Show_Home_Page();
+            fileDataGrid.Rows.Clear();
+        }
+
+        private void Btn_OpenFileLocation_Click(object sender, EventArgs e)
+        {
+            if (filePathConversion == null)
+                Process.Start("explorer.exe");
+            else
+                Process.Start(filePathConversion);
+        }
+        private void fileDataGrid_CellStateChanged(object sender, DataGridViewCellStateChangedEventArgs e)
+        {
+            
+            DataGridViewElementStates state = e.StateChanged;
+            DataGridViewSelectedCellCollection SelectedFiles = fileDataGrid.SelectedCells;
+
+
+            int numFilesRemove = SelectedFiles.Count;
+            if (SelectedFiles.Contains(fileDataGrid[0, fileDataGrid.RowCount-1]))
+                numFilesRemove--;
+
+            //if (SelectedFiles.Contains(fileDataGrid.Rows[fileDataGrid.RowCount - 1].Cells[1]))
+            //    numFilesRemove--;
+            btn_Remove.Text = $"Delete ({numFilesRemove}) Files";
+        }
+
+        private void Btn_Format_Click(object sender, EventArgs e)
+        {
+            Show_Format_Page();
+        }
+
+        private void Btn_Spaced_Click(object sender, EventArgs e)
+        {
+            // Pseudo Radio Button Styling
+            isSingleSpaced = !isSingleSpaced;
+            if (isSingleSpaced)
             {
-                File.Copy("style.css", cssFilename);
+                this.Btn_SingleSpaced.BackColor = System.Drawing.Color.White;

```

Comment:
All of these casts are unnecessary and there are a whole lot of unneeded parentheses

Reply:
You might also want to store the colors instead of creating the color every time you want to use it.

Reply:
Stored colors as privates in Main.Designer

Reply:
Much, much easier to read.